cfgov/teachers_digital_platform/models/activity_index_page.py

```diff
@@ -0,0 +1,528 @@
+                content=search_query).order_by('-_score', '-date')
+        else:
+            sqs = sqs.order_by('-date')
+
+        # Get all facets and their counts
+        facet_counts = sqs.facet_counts()
+        all_facets = self.get_all_facets(
+            FACET_MAP, sqs, facet_counts, facet_queries, selected_facets
+        )
+
+        # List all facet blocks that need to be expanded
+        always_expanded = {'building_block', 'topic', 'school_subject'}
+        conditionally_expanded = {
+            facet_name for facet_name, facet_items in all_facets.items()
+            if any(
+                facet['selected'] is True for facet in facet_items
+            )
+        }
+        expanded_facets = always_expanded.union(set(conditionally_expanded))
+
+        payload.update({
+            'facet_counts': facet_counts,
+            'all_facets': all_facets,
+            'expanded_facets': expanded_facets,
+        })
+
+        # Apply all the active facet values to our search results
+        for facet_narrow_query in facet_queries.values():
+            sqs = sqs.narrow(facet_narrow_query)
+
+        results = [activity_card_map[activity.pk] for activity in sqs]
+        total_results = len(results)
+
+        payload.update({
+            'results': results,
+            'total_results': total_results,
+        })
+        self.results = payload
+        results_per_page = validate_results_per_page(request)
+        paginator = Paginator(payload['results'], results_per_page)
+        current_page = validate_page_number(request, paginator)
+        paginated_page = paginator.page(current_page)
+
+        context_update = {
+            'facet_counts': facet_counts,
+            'facets': all_facets,
+            'activities': paginated_page,
+            'total_results': total_results,
+            'results_per_page': results_per_page,
+            'current_page': current_page,
+            'paginator': paginator,
+            'show_filters': bool(facet_queries),
+        }
+        return context_update
+
+    def get_context(self, request, *args, **kwargs):
+        if not self.activity_setups:
+            self.activity_setups = get_activity_setup()
+        if flag_enabled('ELASTICSEARCH_DSL_TDP'):
+            context_update = self.dsl_search(request, *args, **kwargs)
+        else:
+            context_update = self.haystack_search(request, *args, **kwargs)
+        context = super(ActivityIndexPage, self).get_context(request)
+        context.update(context_update)
+        return context
+
+    # TODO: Remove this function when we switch to DSL
+    def get_all_facets(self, facet_map, sqs, facet_counts, facet_queries, selected_facets):  # noqa: E501
+        all_facets = {}
+        if 'fields' in facet_counts:
+            for facet, facet_config in facet_map:
+                class_object, is_nested = facet_config
+                all_facets_sqs = sqs
+                other_facet_queries = [
```

Comment:
@higs4281 A bit of functionality thats been lost with the new elasticsearch solution is, determining the available facets for a group, based on the selected facets in other groups. That would explain why I've been seeing more facets that return no results when testing this branch locally.